src/ferui-components/tree-view/tree-view-component.ts

```diff
@@ -0,0 +1,399 @@
+      <fui-virtual-scroller
+        [hidden]="loading || error"
+        #scroll
+        class="fui-virtual-scroller"
+        [items]="scrollViewArray"
+        [bufferAmount]="BUFFER_AMOUNT"
+      >
+        <fui-tree-node
+          *ngFor="let node of scroll.viewPortItems"
+          [node]="node"
+          [theme]="colorTheme"
+          [borders]="hasBorders"
+          [dataRetriever]="dataRetriever"
+          (onNodeEvent)="nodeEvent($event)"
+        ></fui-tree-node>
+      </fui-virtual-scroller>
+      <clr-icon *ngIf="loading" class="fui-loader fui-loader-animation" shape="fui-spinner"></clr-icon>
+      <div *ngIf="error" class="error">
+        <clr-icon class="fui-error-icon" shape="fui-error" aria-hidden="true"></clr-icon>
+        <p>Couldn't load content</p>
+      </div>
+    </div>
+  `,
+  styles: [
+    `
+      #fui-tree-view-component {
+        padding: 10px;
+      }
+      .fui-loader {
+        color: #87a1b5;
+        position: relative;
+        top: 50%;
+        left: 50%;
+      }
+      .error {
+        position: relative;
+        top: 50%;
+        left: 50%;
+        color: #87a1b5;
+        text-align: center;
+        transform: translateX(-50%);
+      }
+      .DARK_BLUE {
+        background: #252a3a;
+        color: #fff;
+      }
+      .LIGHT_BLUE {
+        background: #03a6ff;
+        color: #fff;
+      }
+      .GRAY {
+        background: #f5f8f9;
+        color: #252a3a;
+      }
+      .WHITE {
+        background: #fff;
+        color: #252a3a;
+      }
+      .fui-virtual-scroller {
+        height: 100%;
+        width: 100%;
+      }
+      :host(.show-border) #fui-tree-view-component {
+        border: #e5eaee 1px solid;
+        padding: 0;
+      }
+    `,
+  ],
+})
+export class FuiTreeViewComponent<T> implements OnInit, OnDestroy {
+  @Input() treeNodeData: TreeNodeData<T>;
+
+  @Input() dataRetriever: TreeNodeDataRetriever<T> | PagedTreeNodeDataRetriever<T>;
+
+  @Input() config: TreeViewConfiguration;
+
+  @Input() loading: boolean = false;
+
+  @Input() error: boolean = false;
+
+  @HostBinding('class.show-border') border: boolean;
+
+  @ViewChild(FuiVirtualScrollerComponent) vs: FuiVirtualScrollerComponent;
+
+  @ViewChildren(FuiTreeNodeComponent) children: QueryList<FuiTreeNodeComponent<T>>;
+
+  @Output() onNodeEvent: EventEmitter<TreeViewEvent<any>> = new EventEmitter<TreeViewEvent<any>>();
+
+  treeViewStyles: FuiTreeViewComponentStyles;
+  colorTheme: TreeViewColorTheme;
+  scrollViewArray: TreeNode<T>[];
+  hasBorders: boolean;
+  BUFFER_AMOUNT: number = 1000;
+  private rootNode: TreeNode<T>;
+  private nonRootArray: TreeNode<T>[];
+  private scrollObservable: Observable<any>;
+  private scrollSubscription: Subscription;
+  private indexChangeSubscription: Subscription;
+  private originalWidth: number;
+  private scrollPromise: boolean = false;
+  private SERVER_SIDE_COMPONENT: boolean = false;
```

Comment:
please, use normal case for variables that are part of a class. The SCREAMING_SNAKE_CASE is reserved for constants internal to FerUI and shared between multiple components or for enums.

Reply:
Fixed!